tests/php/src/Cli/ReferenceSiteImportCommand.php

```diff
@@ -0,0 +1,282 @@
+	 *
+	 * [--empty-extensions]
+	 * : Empty the extensions folder (plugins & themes except for the AMP & WordPress Importer plugin) before importing the reference content.
+	 *
+	 * [--empty-options]
+	 * : Empty the site options before importing the reference content and only leave default WordPress options in place.
+	 *
+	 * [--skip-site-meta]
+	 * : Skip importing the site meta information like blog name or description.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Import content from a reference site definition file
+	 *     $ wp amp reference-site import example-site
+	 *     Starting the import process...
+	 *     Processing post #1 ("Hello world!") (post_type: post)
+	 *     -- 1 of 1
+	 *     -- Tue, 21 Jun 2016 05:31:12 +0000
+	 *     -- Imported post as post_id #1
+	 *     Success: Finished importing from 'example-site' file.
+	 *
+	 * @when after_wp_load
+	 */
+	public function __invoke( $args, $assoc_args ) {
+		try {
+			( new WpImporterCompat() )->prepare_import();
+		} catch ( Exception $exception ) {
+			WP_CLI::error( 'Unable to import reference site: ' . $exception->getMessage() );
+		}
+
+		$storage = new StorageClient();
+		$storage->registerStreamWrapper();
+
+		list( $site_definition_file ) = $args;
+		$empty_content                = Utils\get_flag_value( $assoc_args, 'empty-content', false );
+		$empty_uploads                = Utils\get_flag_value( $assoc_args, 'empty-uploads', false );
+		$empty_extensions             = Utils\get_flag_value( $assoc_args, 'empty-extensions', false );
+		$empty_options                = Utils\get_flag_value( $assoc_args, 'empty-options', false );
+		$skip_site_meta               = Utils\get_flag_value( $assoc_args, 'skip-site-meta', false );
+
+		if ( 0 !== substr_compare( $site_definition_file, '.json', -5 ) ) {
+			$site_definition_file .= '.json';
+		}
+
+		if ( ! path_is_absolute( $site_definition_file ) ) {
+			$site_definition_file = ReferenceSiteCommandNamespace::REFERENCE_SITES_ROOT . $site_definition_file;
+		}
+
+		if ( ! file_exists( $site_definition_file )
+			|| ! is_readable( $site_definition_file ) ) {
+			WP_CLI::error( "The provided site definition file '{$site_definition_file}' could not be read." );
+		}
+
+		WP_CLI::log( 'Starting the reference site import process...' );
+
+		try {
+			$site_definition = $this->load_site_definition( $site_definition_file );
+		} catch ( Exception $exception ) {
+			WP_CLI::error( "The provided site definition file '{$site_definition_file}' could not be parsed: {$exception->getMessage()}" );
+		}
+
+		if ( $empty_extensions ) {
+			$this->empty_extensions();
+		}
+
+		if ( $empty_content ) {
+			$this->empty_site( $empty_uploads );
+		}
+
+		if ( $empty_options ) {
+			$this->empty_options();
+		}
+
+		WP_CLI::log(
+			'Importing reference site: '
+			. WP_CLI::colorize( "%y{$site_definition->get_name()}%n (%gv{$site_definition->get_version()}%n)" )
```

Comment:
`$site_definition` will be undefined if the try/catch above it's declared in throws an error.

Reply:
The `WP_CLI::error()` ends with an `exit()` and stops the process.